test/include/dlaf_test/matrix/util_tile_blas.h

```diff
@@ -22,38 +24,46 @@ namespace dlaf {
 namespace matrix {
 namespace test {
 
-/// Sets the elements of the tile.
-///
-/// The (i, j)-element of the tile is set to el({i, j}) if op == NoTrans,
-///                                          el({j, i}) if op == Trans,
-///                                          conj(el({j, i})) if op == ConjTrans.
-/// @pre el argument is an index of type const TileElementIndex& or TileElementIndex,
-/// @pre el return type should be T.
-template <class T, class Func>
-void set(Tile<T, Device::CPU>& tile, Func el, blas::Op op) {
+namespace internal {
+template <class ElementGetter>
+auto opValFunc(ElementGetter val, const blas::Op op) {
+  std::function<decltype(val(std::declval<TileElementIndex>()))(TileElementIndex)> op_val;
   switch (op) {
     case blas::Op::NoTrans:
-      set(tile, el);
+      op_val = [&val](TileElementIndex i) { return val(i); };
```

Comment:
I'm wondering if this was/is an error, since we are capturing by reference a variable with function scope inside a lambda, than the lambda outlives the value because it is returned outside of the function, so when it is called the referenced value has been destroyed.

Reply:
Good catch... now it is reference of reference...